packages/keycloak-user-management/src/components/UserList/index.tsx

```diff
@@ -95,11 +95,7 @@ const UserList = (props: UserListTypes): JSX.Element => {
       max: pageSize ?? usersPageSize,
     };
     if (searchParam) {
-      filterParams = {
-        ...filterParams,
-        first: 0,
-        search: searchParam,
-      };
+      filterParams = { search: searchParam };
```

Comment:
I think we need to allow pagination on search incase we have a huge list of users returned from the search query.

Reply:
We could probably implement a recursive approach for this

Reply:
in case if we keep this pagination approach even on search result then how we will get total number of search result ? because the paginated result will only return/fetch the user equal to size of Pagesize we use in query i.e not giving total number of search results we will have

Reply:
Kindly check how keycloak paginates users returned from search query, eg on the first page it returns according to the max page size (default 20),  then when a user clicks on the next page it fetches the next 20 users and so on.

Reply:
We can use an offset-based pagination based on page size and page number. Incremented on page navigation
`https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage/users?first={(pageSize*pageNumber)-pageSize}&max=pageSize&search=opensrp`

Reply:
According to [this](https://stackoverflow.com/questions/66754404/keycloak-does-keycloak-have-any-endpoint-to-get-list-of-user-paginated) the total records count will be returned in the response headers

Reply:
@machariamuguku can you give me example endoint for total number of search results ?

Reply:
@ashfahan I'm not sure but the answer seems to suggest something like this https://github.com/opensrp/web/blob/afb778ca7c47a45453a6c989b2bd01d397e1383c/packages/keycloak-user-management/src/components/forms/UserForm/utils.tsx#L26

Reply:
@ashfahan after an embarrassingly long time I've found the `/users/count` endpoint. [Example](https://keycloak-stage.smartregister.org/auth/admin/realms/opensrp-web-stage/users/count?search=test). It also [takes the search query param](https://github.com/keycloak/keycloak/blob/17d41c472b78b2053165669bc690509bf761e92c/services/src/main/java/org/keycloak/services/resources/admin/UsersResource.java#L333) and returns an integer of number of users matching query